pydantic/schema.py

```diff
@@ -984,7 +984,7 @@ def encode_default(dft: Any) -> Any:
         return dft.value
     elif isinstance(dft, (int, float, str)):
         return dft
-    elif sequence_like(dft):
+    elif isinstance(dft, (list, tuple)):
```

Comment:
I'm really not into the code base - so this is just a non-constructive critisism, but this change would narrow the scope compared to the original sequence_like check. What happens when these instances (set, frozenset, ...) are passed in?

Reply:
I'm also very new to the codebase, but I think the `sequence_like` usage is not very accurate here.

If you look further, after these conditions it will try to use the pydantic_encoder for all unhandled types which in this case is set and frozenset.
And also there are tests for all cases and new ones for set and frozenset, so I think this is not narrowing down any type, just setting up different encoders for these types.

Reply:
sounds reasonable! now we only need @samuelcolvin or @PrettyWood to approve :-)